zanata-war/src/main/java/org/zanata/action/ValidateEmailAction.java

```diff
@@ -60,6 +61,11 @@
     @In
     EmailChangeService emailChangeService;
 
+    @Create
+    public void onCreate() {
+        identity.checkLoggedIn();
```

Comment:
On 2015-07-31 13:18, Carlos Munoz wrote:
> In zanata-war/src/main/java/org/zanata/action/ValidateEmailAction.java
> <https://github.com/zanata/zanata-server/pull/908#discussion_r35943963>:
> 
>> @@ -60,6 +61,11 @@
>>      @In
>>      EmailChangeService emailChangeService;
>>  
>> +    @Create
>> +    public void onCreate() {
>> +        identity.checkLoggedIn();
> 
> So, isn't this line doing exactly that?

It is, it's just not very declarative.


> I thought a failed |<restrict>|
> in |pages.xml| would result in an authorization exception as well.

I think it will, for now, but we're trying to move away from it in
preparation for Seam->CDI.

> But, to answer your question, nothing really comes to mind. CDI
> interceptors can be placed at the class level, in which case they will
> apply to all methods, but I'm fairly certain this doesn't include the
> creation of the bean.

This page mentions @PostConstruct for interceptor classes:
  https://docs.oracle.com/javaee/6/tutorial/doc/gkigq.html

and this one mentions @PostCreate for interceptor classes:
  http://docs.oracle.com/javaee/6/tutorial/doc/gkedm.html

but I'm not sure if we can use either or both of them with plain CDI.

-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat